### PR TITLE
Refactor supermuc documentation page

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
In this PR, I refactor the supermuc documentation page, now detailing how to install the seissol spack package.
I nevertheless keep information about how to compile seissol-env.

On NG, there are two user_spack packages. 
One with spack 0.17.1 (extremly old) and another with spack 0.22.0.dev0.

seissol-env was installed with  0.17.1.
It is tricky to install the new seissol package based on this version (too modern).
On the other hand, I have trouble getting workable modules with spack 0.22.0.dev0.
E.g. module of dependencies do not get created (easi), even if I manually try to create them (it works on heisenbug).
Anyway, this is not such a big issue (e.g. the easi python module I was needing can be using by manually adding the path to the library to PYTHONPATH).
That's why I still detail the seissol-env installation, with spack 0.17.1.

### CODE REVIEW

File: Documentation/supermuc.rst [hunk 1]
```diff
@@ -36,100 +29,36 @@ Load module. Add these lines to .bashrc:
...
-
-.. code-block:: yaml
-
-    modules:
-      default:
-        roots:
-         tcl: your_custom_path_2_modules
 
-This can be useful to share packages with other user of a SuperMUC project.
+Install pspamm (see :ref:`pypi_behind_firewall` for the proxy):
```
Comment (davschneller) [inline Documentation/supermuc.rst:33]: Is it maybe useful to also insert a comment about `--no-build-isolation` into that article?

Comment (davschneller) [reply Documentation/supermuc.rst:33]: (we could also recommend/suggest using a virtual environment for the Python packages, if it helps—but maybe that's a bit out of scope for this topic here)

File: Documentation/supermuc.rst [hunk 2]
```diff
@@ -196,3 +125,170 @@ Many post-processing scripts of SeisSol require Python dependencies.
 We describe how to use pip on SuperMUC at see :ref:`pypi_behind_firewall`.
 
 
+Using Sanitizer
```
Comment (davschneller) [inline Documentation/supermuc.rst:128]: `Using the Sanitizer` ? (the article would seem more natural to me here)

File: Documentation/supermuc.rst [hunk 3]
```diff
@@ -196,3 +125,170 @@ Many post-processing scripts of SeisSol require Python dependencies.
...
+Note that to use sanitizer (https://en.wikipedia.org/wiki/AddressSanitizer), SeisSol needs to be compiled with gcc.
+For that modules and compiler need to be switched:
+
+.. code-block:: bash
+
+    module switch seissol-env seissol-env/develop-gcc11
+
+    export CC=mpigcc
+    export CXX=mpigxx
+    export FC=mpifc
```
Comment (davschneller) [inline Documentation/supermuc.rst:140]: `FC` is not needed anymore—also `gcc` and `gxx` should be enough actually—CMake will do the rest automatically.

File: Documentation/supermuc.rst [hunk 4]
```diff
@@ -196,3 +125,170 @@ Many post-processing scripts of SeisSol require Python dependencies.
 We describe how to use pip on SuperMUC at see :ref:`pypi_behind_firewall`.
 
 
+Using Sanitizer
+---------------
+
+Note that to use sanitizer (https://en.wikipedia.org/wiki/AddressSanitizer), SeisSol needs to be compiled with gcc.
```
Comment (davschneller) [inline Documentation/supermuc.rst:131]: It's also possible to use clang, iirc; but one of the arguments doesn't work then (the `static-libasan` iirc)

File: Documentation/supermuc.rst [hunk 5]
```diff
@@ -196,3 +125,170 @@ Many post-processing scripts of SeisSol require Python dependencies.
...
+        - spec: python@3.8.11
+          buildable: False
+          modules:
+           - python/3.8.11-extended
+
+
+Compiling seissol spack package
+-------------------------------
+ 
+The seissol package is similar with the seissol-env package (it gathers all dependencies of seissol), but also compiles a specific version of seissol.
```
Comment (davschneller) [inline Documentation/supermuc.rst:209]: `similar to`

Comment (davschneller) [reply Documentation/supermuc.rst:209]: Also: `but also compiles a specific version of seissol itself` ... or something to highlight that SeisSol is included here

**Comment by Thomas-Ulrich** [comment]
@davschneller: Thanks for the review!

**Pull request merged**